src/mcmc/hmc.jl

```diff
@@ -59,30 +61,23 @@ sample(gdemo([1.5, 2]), HMC(0.1, 10), 1000)
 sample(gdemo([1.5, 2]), HMC(0.01, 10), 1000)
 ```
 """
-struct HMC{AD, space, metricT <: AHMC.AbstractMetric} <: StaticHamiltonian{AD}
+struct HMC{AD, space, metricT <: AHMC.AbstractMetric} <: StaticHamiltonian
     ϵ::Float64 # leapfrog step size
     n_leapfrog::Int # leapfrog step number
+    adtype::AD
 end
 
-HMC(args...; kwargs...) = HMC{ADBackend()}(args...; kwargs...)
-function HMC{AD}(ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple) where {AD, metricT <: AHMC.AbstractMetric}
-    return HMC{AD, space, metricT}(ϵ, n_leapfrog)
-end
-function HMC{AD}(
-    ϵ::Float64,
-    n_leapfrog::Int,
-    ::Tuple{};
-    kwargs...
-) where AD
-    return HMC{AD}(ϵ, n_leapfrog; kwargs...)
+function HMC(ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple; adtype::ADTypes.AbstractADType = ADBackend()) where {metricT <: AHMC.AbstractMetric}
```

Comment:
Should we remove this constructor and only support `metricT` as keyword argument? Or make all arguments keyword arguments?

Generally, these HMC constructors are quite messy...